lib_eio/core/eio__core.mli

```diff
@@ -286,6 +286,39 @@ module Fiber : sig
   (** [iter f x] is like [List.iter f x] except that the invocations of [f] are
       run concurrently in separate fibers.
       @param max_fibers Maximum number of fibers to run concurrently *)
+
+  (** {2 Fiber-local variables}
+
+      Each fiber maintains a map of additional variables associated with it,
+      which can be used to store fiber-related state or context. This map is
+      propagated to any forked fibers.
+
+      While fiber-local variables can be useful, they can also make code much
+      harder to reason about, as they effectively act as another form of global
+      state. When possible, prefer passing arguments around explicitly. *)
+
+  type 'a key
+  (** ['a key] is a fiber-local variable of type ['a]. *)
```

Comment:
```suggestion
  (** ['a key] is a fiber-local variable of type ['a].
      
      Since the key is required to get or set a variable,
      a library can keep its key private to control how the variable can be accessed. *)
```